src/pkg/packager/create.go

```diff
@@ -467,3 +478,101 @@ func generatePackageChecksums(basePath string) (string, error) {
 	// Calculate the checksum of the checksum file
 	return utils.GetSHA256OfFile(checksumsFilePath)
 }
+
+// loadDifferentialData extracts the zarf config of a designated 'reference' package that we are building a differential over and creates a list of all images and repos that are in the reference package
+func (p *Packager) loadDifferentialData() error {
+	if p.cfg.CreateOpts.DifferentialData.DifferentialPackagePath == "" {
+		return nil
+	}
+
+	tmpDir, _ := utils.MakeTempDir("")
+	defer os.RemoveAll(tmpDir)
+
+	// Load the package spec of the package we're using as a 'reference' for the differential build
+	var differentialZarfConfig types.ZarfPackage
+	if err := archiver.Extract(p.cfg.CreateOpts.DifferentialData.DifferentialPackagePath, config.ZarfYAML, tmpDir); err != nil {
+		return fmt.Errorf("unable to extract the differential zarf package spec: %w", err)
+	}
+	if err := utils.ReadYaml(filepath.Join(tmpDir, config.ZarfYAML), &differentialZarfConfig); err != nil {
+		return fmt.Errorf("unable to load the differential zarf package spec: %w", err)
+	}
+
+	// Generate a list of all the images and repos that are included in the provided package
+	allIncludedImages := []string{}
+	allIncludedRepos := []string{}
+	for _, component := range differentialZarfConfig.Components {
+		allIncludedImages = append(allIncludedImages, component.Images...)
+		allIncludedRepos = append(allIncludedRepos, component.Repos...)
+	}
+	p.cfg.CreateOpts.DifferentialData.DifferentialImages = allIncludedImages
+	p.cfg.CreateOpts.DifferentialData.DifferentialRepos = allIncludedRepos
+	p.cfg.CreateOpts.DifferentialData.DifferentialPackageVersion = differentialZarfConfig.Metadata.Version
+
+	if differentialZarfConfig.Metadata.Version == p.cfg.Pkg.Metadata.Version {
+		message.Warnf("You are creating a differential package with the same version as the package you are using as a reference. This is not recommended.")
```

Comment:
Should we throw a `continue?` prompt here as this is a pretty major thing we want to prevent

Reply:
We may want to explicitly error out and force them to change a version.  We could add something to the error message like: `If you are building a patch or development build you must increment the patch number or add a development suffix (i.e. 1.0.0+dev1)`